src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/dcoff.cpp

```diff
@@ -21,6 +21,97 @@
 namespace ov {
 namespace npuw {
 
+namespace pattern_utils {
+
+std::shared_ptr<ov::op::v0::MatMul> find_matmul_downwards(const std::shared_ptr<ov::Node>& start_node) {
+    std::shared_ptr<ov::Node> current_node = start_node;
+    while (current_node) {
+        // Check if the current node is a MatMul
+        if (auto matmul = std::dynamic_pointer_cast<ov::op::v0::MatMul>(current_node)) {
+            return matmul;
+        }
+        // Move to the next node in the path if there is one
+        if (!current_node->outputs().empty()) {
+            auto output = current_node->outputs().at(0);
+            if (!output.get_target_inputs().empty()) {
+                current_node = output.get_target_inputs().begin()->get_node()->shared_from_this();
+            } else {
+                // No further outputs, end the search
+                break;
+            }
+        } else {
+            // No outputs, end the search
+            break;
+        }
+    }
+    return nullptr; // MatMul not found
+}
+
+
+std::shared_ptr<ov::op::v0::MatMul> get_root_matmul(ov::pass::pattern::Matcher& m) {
+    auto& node_to_output = m.get_pattern_value_map();
+
+    // If the map is not empty, start the search from the first matched node
+    if (!node_to_output.empty()) {
+        auto start_node = node_to_output.begin()->second.get_node_shared_ptr();
+        return find_matmul_downwards(start_node);
+    }
+
+    // If the map is empty or no MatMul node is found, return nullptr
+    LOG_DEBUG("NO MATMUL FOUND!");
+    return nullptr;
+}
+
+bool transpose_required(const std::shared_ptr<ov::op::v0::MatMul>& matmul_node) {
+    if (!matmul_node) {
+        LOG_DEBUG("NOT a MATMUL NODE!");
+    }
```

Comment:
Just place an assert that the pointer is not null. Also, who'd pass a null pointer here?